systemtest/src/test/java/io/strimzi/systemtest/operators/FeatureGatesST.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.systemtest.operators;
+
+import io.fabric8.kubernetes.api.model.ContainerPort;
+import io.fabric8.kubernetes.api.model.EnvVar;
+import io.fabric8.kubernetes.api.model.Pod;
+import io.fabric8.kubernetes.api.model.apps.Deployment;
+import io.fabric8.kubernetes.api.model.apps.DeploymentBuilder;
+import io.strimzi.systemtest.AbstractST;
+import io.strimzi.systemtest.Constants;
+import io.strimzi.systemtest.Environment;
+import io.strimzi.systemtest.annotations.ParallelNamespaceTest;
+import io.strimzi.systemtest.resources.kubernetes.DeploymentResource;
+import io.strimzi.systemtest.resources.operator.BundleResource;
+import io.strimzi.systemtest.templates.crd.KafkaTemplates;
+import io.strimzi.systemtest.utils.kubeUtils.objects.PodUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtensionContext;
+
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/**
+ * Feature Gates should give us additional options on
+ * how to control and mature different behaviors in the operators.
+ * https://github.com/strimzi/proposals/blob/main/022-feature-gates.md
+ */
+public class FeatureGatesST extends AbstractST {
+
+    static final String NAMESPACE = "feature-gates-tests";
+    private static final Logger LOGGER = LogManager.getLogger(FeatureGatesST.class);
+
+    private Deployment clusterOperator;
+    private List<EnvVar> defaultEnvVars;
+    private List<EnvVar> testEnvVars;
+
+    @BeforeAll
+    void setup(ExtensionContext extensionContext) {
+        installClusterWideClusterOperator(extensionContext, NAMESPACE, Constants.CO_OPERATION_TIMEOUT_DEFAULT, Constants.RECONCILIATION_INTERVAL);
+        clusterOperator = DeploymentResource.getDeploymentFromYaml(BundleResource.PATH_TO_CO_CONFIG);
+        defaultEnvVars = clusterOperator.getSpec().getTemplate().getSpec().getContainers().get(0).getEnv();
+    }
+
+    @AfterEach
+    void afterEach(ExtensionContext extensionContext) {
+        clusterOperator = redeployCOWithEnvVars(defaultEnvVars);
+        testEnvVars = null;
+    }
+
+    @BeforeEach
+    void beforeEachSetClusterOperator(ExtensionContext extensionContext) {
+        clusterOperator = DeploymentResource.getDeploymentFromYaml(BundleResource.PATH_TO_CO_CONFIG);
+        testEnvVars = clusterOperator.getSpec().getTemplate().getSpec().getContainers().get(0).getEnv();
+    }
```

Comment:
I know it makes sense to have these `@BeforeAll`, `@BeforeEach` etc. on the beginning of the class, but right now we have it at the end of all ST classes, can you please move it there to follow this "pattern"?